lib/osf-components/addon/components/editable-field/provider-metadata-manager/component.ts

```diff
@@ -0,0 +1,104 @@
+import { tagName } from '@ember-decorators/component';
+import Component from '@ember/component';
```

Comment:
Can we make this a glimmer component?

Reply:
Tricky. We don't have anything using the management pattern as a glimmer component. When I convert it to glimmer, it doesn't like the `@layout` decorator, and removing that causes this (and the other components if I try to convert them) to not render. Do you happen to know what's going wrong here?

Reply:
Glimmer components doesn't need the `@layout` decorator.  What you need in turn is to add a `template.js` re-export in `app/editable-field/provider-metadata-manager` and it should work.

Reply:
Still not rendering, and there are no errors. Any other things you can think of?

Reply:
@brianjgeiger, glimmer components only support `constructor` and `willDestroy` as lifecycle hooks. So whatever is being done in the `didReceiveAttrs` is ignored and should be done using render modifiers i.e `{{did-insert}}` and/or `{{did-update}}` See https://guides.emberjs.com/release/upgrading/current-edition/glimmer-components/#toc_lifecycle-hooks--modifiers

Reply:
I think you also need to refer to arguments passed to the glimmer component using `this.args.<>`. For example, the `registration` argument would be `this.args.registration` instead of `this.registration`.

Reply:
Well, it _almost_ renders.

<img width="834" alt="Screen Shot 2021-03-23 at 11 29 33 AM" src="https://user-images.githubusercontent.com/6599111/112172565-33e1f380-8bcb-11eb-8bb7-5ae66c828686.png">